example/src/Main.purs

```diff
@@ -27,17 +29,76 @@ import Ace.UndoManager as UndoManager
 import Ace.VirtualRenderer as VirtualRenderer
 import Ace.Ext.LanguageTools as LanguageTools
 import Ace.Ext.LanguageTools.Completer as Completer
+import Ace.KeyBinding as KeyBinding
 
 foreign import rules :: Rules
 foreign import onLoad :: forall e. Eff e Unit -> Eff e Unit
 
-main :: Eff (ace :: ACE, console :: CONSOLE) Unit
+main :: forall e. Eff (ref ::REF, console :: CONSOLE, ace :: ACE, dom :: DOM|e) Unit
 main = onLoad $ do
-
   Config.set Config.basePath "foo"
 
   -- Create an editor
   editor <- Ace.edit "editor" ace
+  session <- Editor.getSession editor
+  document <- Session.getDocument session
+  Editor.setValue "blablabla \n tr  test boo boo" Nothing editor
+
+  startAnchor <- Document.createAnchor 1 5 document
+  endAnchor <- Document.createAnchor 1 9 document
+  Anchor.setInsertRight true endAnchor
+
+  range <- Range.create 1 4 1 10
+  markerRef <-
+    Session.addMarker
+      range
+      "readonly-highlight"
+      "string"
+      false
+      session
+    >>= newRef
+
+  let rerenderMarker _ = do
+        readRef markerRef >>= flip Session.removeMarker session
+        Position {row: startRow, column: startColumn}
+          <- Anchor.getPosition startAnchor
+        Position {row: endRow, column: endColumn}
+          <- Anchor.getPosition endAnchor
+        markRange <- Range.create
+                       startRow
+                       (startColumn - one)
+                       endRow
+                       (endColumn + one)
+        newMId <- Session.addMarker
+                  markRange
+                  "readonly-highlight"
+                  "string"
+                  false
+                  session
+
+        writeRef markerRef newMId
+        pure unit
+
+  Anchor.onChange startAnchor rerenderMarker
```

Comment:
It's important to rerender marker in handler of anchor position change, not in keyboard handler because of event handlers call order.